kube-client/src/config/mod.rs

```diff
@@ -206,13 +206,42 @@ impl Config {
         Ok(config)
     }
 
-    /// Create configuration from the cluster's environment variables
-    ///
-    /// This follows the standard [API Access from a Pod](https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/#accessing-the-api-from-a-pod)
-    /// and relies on you having the service account's token mounted,
-    /// as well as having given the service account rbac access to do what you need.
+    /// Replaced by [`Self::load_in_cluster`].
+    #[deprecated(since = "0.75.0", note = "use Config::load_in_cluster")]
     pub fn from_cluster_env() -> Result<Self, InClusterError> {
-        let cluster_url = incluster_config::kube_dns();
+        Self::load_in_cluster()
+    }
```

Comment:
`Config::from_cluster_env` is a misleading name. Rather than change it to actually use the environment, it seems better to deprecate it and replace it with `load_in_cluster`. I thought _load_ helps to indicate that it does things like read from disk. _in_cluster_ matches the terminology in client-go.